### PR TITLE
Fix for static analysis issues

### PR DESCRIPTION
**Pull request by simonami99**
Below are the issues fixed:
- Resolved BUFFER_SIZE issues
- Resolved STRING_NULL issues

Test done:
- Successfully verified

Tracked-On : OAM-122341
Signed-off-by: Simona Mishra <simona.mishra@intel.com>

### CODE REVIEW

**Review by akodanka — APPROVED** [review]
LGTM

File: drm/DrmAtomicStateManager.cpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ auto DrmAtomicStateManager::CommitFrame(AtomicCommitArgs &args) -> int {
     return -ENOMEM;
   }
 
-  int out_fence = -1;
+  int out_fence = 0;
```
Comment (akodanka) [inline drm/DrmAtomicStateManager.cpp:76]: Can you double check how this change fixed any issue?
Is the below function actually expecting the address of a local(stack) variable as a uint64_t ?

**Review by akodanka — APPROVED** [review]
LGTM
Just validate the query raised in the review comment before submitting.

File: drm/DrmAtomicStateManager.cpp (hunk 1 above)
Comment (simonami99) [reply drm/DrmAtomicStateManager.cpp:76]: The issue found was: "out_fence" is passed to a parameter that cannot be negative -> for line 212
Initializing it to 0 resolves the issue.

File: drm/DrmAtomicStateManager.cpp [hunk 2]
```diff
@@ -234,9 +234,9 @@ PresentTrackerThread::PresentTrackerThread(DrmAtomicStateManager *st_man)
   pt_ = std::thread(&PresentTrackerThread::PresentTrackerThreadFn, this);
 }
 
-PresentTrackerThread::~PresentTrackerThread() {
-  ALOGI("PresentTrackerThread successfully destroyed");
-}
+PresentTrackerThread::~PresentTrackerThread() noexcept {
+	ALOGI("PresentTrackerThread successfully destroyed");
+ }
```
Comment (JaikrishnaNemallapudi) [inline drm/DrmAtomicStateManager.cpp:239]: removd space before }

File: drm/DrmDevice.cpp [hunk 3]
```diff
@@ -49,7 +49,7 @@ auto DrmDevice::CreateInstance(std::string const &path,
   return device;
 }
 
-DrmDevice::DrmDevice(ResourceManager *res_man) : res_man_(res_man) {
+DrmDevice::DrmDevice(ResourceManager *res_man) : res_man_(res_man),preferred_mode_limit_(false),planes_enabling_(false),planes_num_(0),color_adjustment_enabling_(false){
```
Comment (JaikrishnaNemallapudi) [inline drm/DrmDevice.cpp:52]: add space after each ,

File: drm/DrmFbImporter.cpp [hunk 4]
```diff
@@ -93,7 +93,7 @@ auto DrmFbIdHandle::CreateInstance(BufferInfo *bo, GemHandle first_gem_handle,
   return local;
 }
 
-DrmFbIdHandle::~DrmFbIdHandle() {
+DrmFbIdHandle::~DrmFbIdHandle() noexcept{
```
Comment (JaikrishnaNemallapudi) [inline drm/DrmFbImporter.cpp:96]: add space between noexcept and {

File: hwc3-server/default/impl/HwcLoader.cpp [hunk 5]
```diff
@@ -73,13 +73,14 @@ hwc2_device_t* HwcLoader::openDeviceWithAdapter(const hw_module_t* module, bool*
...
 hwc2_device_t* HwcLoader::adaptGrallocModule(const hw_module_t* module) {
-    framebuffer_device_t* device;
+    framebuffer_device_t* device = nullptr;
     int error = framebuffer_open(module, &device);
     if (error) {
         ALOGE("failed to open framebuffer device: %s", strerror(-error));
         return nullptr;
     }
 
+
```
Comment (JaikrishnaNemallapudi) [inline hwc3-server/default/impl/HwcLoader.cpp:83]: remove extra line space

**Review by JaikrishnaNemallapudi — CHANGES_REQUESTED** [review]

**Review by JaikrishnaNemallapudi — APPROVED** [review]
lgtm

**Pull request closed without merging**